grails-app/taglib/org/pih/warehouse/DatePickerTagLib.groovy

```diff
@@ -38,10 +111,28 @@ class DatePickerTagLib {
         def autocomplete = attrs.autocomplete ?: 'off'
         def dataTestId = attrs['data-testid'] ?: 'date-picker'
 
-        if (value) {
-            if (value instanceof Date) {
-                value = (attrs.format && attrs.value) ? new SimpleDateFormat(attrs.format).format(attrs.value) : ""
-            }
+        // If a value was provided on page load, stringify it for the date picker to use as a default. Note that the
+        // date picker uses the javascript Date type (not java.util.Date), which has a different pattern/format rules
+        // and a constructor that defaults to midnight in the server timezone if given date-only (no time or offset).
+        // It's unclear why, but the date string must be formatted to match the 'altFormat' format (MM/dd/yyyy).
+        // Otherwise we get weird behaviour (the selected dates shift by one day every time the page refreshes).
+        String value
+        switch (attrs.value) {
+            case Date:
+                value = Constants.MONTH_DAY_YEAR_DATE_FORMATTER.format(attrs.value)
+                break
+            case LocalDate:
+                value = DateTimeFormatter.ofPattern('MM/dd/yyyy').format(attrs.value as LocalDate)
+                break
+            case String:
```

Comment:
The String case feels fine, but does the value need to be normalized somehow if there is a value provided? Can you make sure to add some test cases for various string values?